reproducers/managed_ptr_multiple_inheritance_reproducer.cpp

```diff
@@ -0,0 +1,602 @@
+
+            /// Removing this makes the bug go away!!!
+            virtual int GetNumStrings() const = 0 ;
+      } ;
+
+      class  Compute {
+         public:
+            CHAI_HOST_DEVICE Compute() {
+#ifdef CHAI_DEVICE_COMPILE
+               printf("Table::Compute::Compute POINTER %p\n", this) ;
+#endif
+            }
+
+            CHAI_HOST_DEVICE virtual ~Compute() {}
+
+         protected:
+
+            CHAI_HOST_DEVICE real8 BaseXFromNewX(const real8 *newX) const ;
+
+            CHAI_HOST_DEVICE virtual real8 RootFromBaseX(real8 baseX, const real8 *vals) const = 0;
+
+         private:
+            Compute(const Compute &other);
+            Compute &operator=(const Compute &other);
+      } ;
+   protected:
+
+      CHAI_HOST_DEVICE int Search(real8 xin, const real8* xvals, int numX) const ;
+
+   private:
+
+      Table(const Table &other);
+
+      Table& operator =(const Table&);
+};
+
+class  SimpleTable : public Table {
+   public:
+      CHAI_HOST_DEVICE SimpleTable() : Table() {
+#ifdef CHAI_DEVICE_COMPILE
+         printf("SimpleTable::SimpleTable POINTER %p\n", this) ;
+#endif
+      }
+
+
+      CHAI_HOST_DEVICE virtual ~SimpleTable() {}
+
+   private:
+      SimpleTable(const SimpleTable& other);
+
+      SimpleTable& operator =(const SimpleTable&);
+};
+
+class  Table1D : public SimpleTable {
+   public:
+      CHAI_HOST_DEVICE Table1D() : SimpleTable() {
+#ifdef CHAI_DEVICE_COMPILE
+         printf("Table1D::Table1D POINTER %p\n", this) ;
+#endif
+      }
+
+      CHAI_HOST_DEVICE virtual ~Table1D() {}
+
+      CHAI_HOST_DEVICE virtual real8 Evaluate(real8 xin) const = 0 ;
+
+   protected:
+      CHAI_HOST_DEVICE Table1D(const Table1D &other) = delete;
+
+      CHAI_HOST_DEVICE Table1D& operator=(const Table1D &other) = delete;
+} ;
+
+class  DataTable1D : public Table1D, public Table::Data {
+   public:
+      CHAI_HOST_DEVICE DataTable1D(const int numX, const real8 *x, const real8 *fx) ;
+
+      CHAI_HOST_DEVICE virtual ~DataTable1D();
+
+      CHAI_HOST_DEVICE virtual real8 Evaluate(real8 xin) const override;
+
+      using Table1D::Evaluate ;
+
+      CHAI_HOST_DEVICE real8 GetXAt(int pos) const;
+
+      CHAI_HOST_DEVICE real8 GetYAt(int pos) const;
+
+   protected:
+      CHAI_HOST_DEVICE virtual real8 innerEvaluate(real8 x) const = 0 ;
+
+      CHAI_HOST_DEVICE int SearchX(real8 xin) const ;
+
+      const int m_numX ;
+
+      real8 *m_x ;
+
+      real8 *m_fx ;
+} ;
+
+class LinearTable1D : public DataTable1D {
+   public:
+      CHAI_HOST_DEVICE  LinearTable1D(const int numX, const real8 *x, const real8 *fx) ;
+
+      CHAI_HOST_DEVICE virtual ~LinearTable1D() ;
+
+      CHAI_HOST_DEVICE virtual real8 innerEvaluate(real8 x) const override;
+
+   private:
+      CHAI_HOST_DEVICE LinearTable1D() = delete;
+      CHAI_HOST_DEVICE LinearTable1D(const LinearTable1D &other) = delete;
+      CHAI_HOST_DEVICE LinearTable1D &operator=(const LinearTable1D &other) = delete;
+
+      real8 *m_x0 ;
```

Comment:
Would it be hard to remove some of these data members?

Reply:
When I tried removing data members (not this one, but most of the others), I was not able to reproduce the error (but it gets confusing whether I messed something up in the removal or not).